app/src/main/java/com/example/hwone/userdto/UserPreferencesDataStore.kt

```diff
@@ -0,0 +1,73 @@
+package com.example.hwone.userdto
+
+import android.content.Context
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import androidx.datastore.preferences.preferencesDataStore
+import com.example.hwone.Constants.EXTRA_EMAIL
+import com.example.hwone.Constants.EXTRA_PASSWORD
+import com.example.hwone.Constants.EXTRA_REMEMBER
+import com.example.hwone.Constants.IS_LOGGED_IN
+import com.example.hwone.Constants.USER_PREFS
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.flow.map
+
+// Extension property for DataStore
+private val Context.dataStore by preferencesDataStore(USER_PREFS)
+
+class UserPreferencesDataStore(private val context: Context) : UserPreferencesInterface {
+
+    private val dataStore = context.dataStore
```

Comment:
Variable created in class is called class field and existing as long as class. Hence no need to hold this field, because you always be holding in memory bigger size than could be.

Reply:
Done